### PR TITLE
Add methods that can throw

### PR DESCRIPTION
**Pull request by SferaDev**
Closes: #559

### CODE REVIEW

File: test/integration/read.test.ts [hunk 1]
```diff
@@ -75,13 +105,30 @@ describe('record read', () => {
...
+    expect(result).toHaveLength(items.length);
+    expect(result).toEqual(items.map(() => null));
+  });
+
   test('read multiple with falsy values, throws', async () => {
+    const items = [null, undefined, false, 0, ''];
+
     // @ts-ignore
-    expect(client.db.teams.read([null, undefined, false, 0, ''])).rejects.toThrow();
+    await expect(client.db.teams.readOrThrow(items)).rejects.toThrow();
```
Comment (SferaDev) [inline test/integration/read.test.ts:132]:
```suggestion
   expect(client.db.teams.readOrThrow(items)).rejects.toThrow();
```

File: test/integration/read.test.ts [hunk 2]
```diff
@@ -75,13 +105,30 @@ describe('record read', () => {
...
+    expect(result).toHaveLength(items.length);
+    expect(result).toEqual(items.map(() => null));
+  });
+
   test('read multiple with falsy values, throws', async () => {
+    const items = [null, undefined, false, 0, ''];
+
     // @ts-ignore
-    expect(client.db.teams.read([null, undefined, false, 0, ''])).rejects.toThrow();
+   expect(client.db.teams.readOrThrow(items)).rejects.toThrow();
```
Comment (SferaDev) [inline test/integration/read.test.ts:132]:
```suggestion
    expect(client.db.teams.readOrThrow(items)).rejects.toThrow();
```

**Comment by SferaDev** [comment]
After a meeting with @tsg we decided:

- `Record` methods like `team.read`, `team.update` or `team.delete`, throw if record has been deleted.
- `getFirst`, `read`, `update`, `delete` are safe, and we create their `orThrows` variants
- `create` is safe but the return type if `id` is not passed doesn't include `null`
  - The safe operation without `null` on `create` is `createOrUpdate`

File: packages/client/src/schema/repository.ts [hunk 3]
```diff
@@ -1039,11 +1440,17 @@ export const initObject = <T>(
...
+        } else {
+          result[column.name] = null;
         }
 
         break;
       }
       default:
+        result[column.name] = value ?? null;
+        if (column.notNull === true && value === null) {
+          console.error(`Parse error, column ${column.name} is non nullable and value resolves null`);
```
Comment (SferaDev) [inline packages/client/src/schema/repository.ts:1452]: I hope this doesn't happen, I considered throwing but wasn't sure. Made it safe in the end...

**Comment by SferaDev** [comment]
With a bit of luck this is the last breaking change that I think we need to introduce with our current features.

File: packages/client/src/schema/repository.ts (hunk 3 above)
Comment (gimenete) [reply packages/client/src/schema/repository.ts:1452]: 👍 to make it safe.

I think at some point we'll need a "verbose/debug" option or an optional "log" function or something like that to decide how to print these messages. But of course, out of the scope of this PR. This is a very edge case and it's fine to leave it here for now.

**Review by gimenete — APPROVED** [review]
Fantastic job 👏 👏 👏  🥇

**Pull request merged**